scripts/street_sign_recognizer.py

```diff
@@ -18,47 +19,127 @@ def __init__(self):
+        cv2.createTrackbar('sat lower bound', 'threshold_image', self.sat_lower_bound, 255, self.set_sat_lower_bound)
+        cv2.createTrackbar('val upper bound', 'threshold_image', self.value_upper_bound, 255, self.set_value_upper_bound)
+        cv2.createTrackbar('hue upper bound', 'threshold_image', self.hue_upper_bound, 255, self.set_hue_upper_bound)
+        cv2.createTrackbar('sat upper bound', 'threshold_image', self.sat_upper_bound, 255, self.set_sat_upper_bound)
+        cv2.createTrackbar('dilate erode iterations', 'threshold_image', self.dilate_erode_iterations, 5, self.set_dilate_erode_iterations)
+
+        #init matcher from other half of project
+        self.matcher = tm.TemplateMatcher(self.images)
+        self.pred = {}
+
+    #callback functions for sliders
+
+    def set_value_lower_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the value lower bound """
+        self.value_lower_bound = val
+    def set_hue_lower_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the hue lower bound """
+        self.hue_lower_bound = val
+    def set_sat_lower_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the saturation lower bound """
+        self.sat_lower_bound = val
+    def set_value_upper_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the value upper bound """
+        self.value_upper_bound = val
+    def set_hue_upper_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the hue upper bound """
+        self.hue_upper_bound = val
+    def set_sat_upper_bound(self, val):
+        """ A callback function to handle the OpenCV slider to select the saturation upper bound """
+        self.sat_upper_bound = val
+    def set_dilate_erode_iterations(self, val):
+        """ A callback function to handle the OpenCV slider to select the number of times to dilate and erode"""
+        self.dilate_erode_iterations = val
 
     def process_image(self, msg):
         """ Process image messages from ROS and stash them in an attribute
-            called cv_image for subsequent processing """
+            called cv_image. Converts to binary, erodes, and dilates for cropping. Calls cropping, then passes to matcher"""
+
         self.cv_image = self.bridge.imgmsg_to_cv2(msg, desired_encoding="bgr8")
+        kernel = np.ones((5,5),np.uint8)
+
+        #filter image
+        self.hsv_image = cv2.cvtColor(self.cv_image, cv2.COLOR_BGR2HSV) #change from rgb to hsv
+        self.binary_image = cv2.inRange(self.hsv_image, (self.hue_lower_bound,self.sat_lower_bound,self.value_lower_bound), (self.hue_upper_bound,self.sat_upper_bound,self.value_upper_bound))
+        self.binary_image = cv2.erode(self.binary_image, kernel, iterations=self.dilate_erode_iterations)
```

Comment:
Nice!  Erode and Dialate make sense as good processing steps to remove the "specks" that appear with a not-so-perfect inRange threshold operation.   Do you mind commenting on what your process was for figuring out to use this method?